java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/TestGenerationUtils.java

```diff
@@ -444,6 +444,17 @@ private static String getDefaultTestFullyQualifiedName(ITypeBinding typeBinding,
         return defaultName;
     }
 
+    private static String getClassName(ITypeBinding typeBinding) {
```

Comment:
How about simply using `typeBinding.getBinaryName().replaceAll("\\$", "_")`?

Reply:
I'm afraid that will not only replace '$' as a splitter, but also '$' as part of the name. (Though that should not be a very common case).

Reply:
> The upstream runner could not fine tests when '$' is in class name,

If the class itself has '$' in its name, does the runner support it?

Reply:
Oh I got your point. Yes if upstream cannot handle this case well, there is no point to keep `$` in the class name.

Reply:
@testforstephen PR updated. new code replaces all the `$` in type name to `_`, but reserve the `$` in packages, since the `$` in packages won't affect the execution.